### PR TITLE
Print more info on image tags and images

### PR DESCRIPTION
**Pull request by smarterclayton**
Support partial retrieval of image stream images by the prefix
of their image name. Allow trimming of the digest as well for
simpler output

Add imageName field to imageStreamImage.

@ncdc

### CODE REVIEW

File: pkg/image/registry/imagestreamimage/rest.go [hunk 1]
```diff
@@ -64,24 +67,40 @@ func (r *REST) Get(ctx kapi.Context, id string) (runtime.Object, error) {
...
-				imageWithMetadata, err := api.ImageWithMetadata(*image)
-				if err != nil {
-					return nil, err
-				}
-				isi := api.ImageStreamImage{Image: *imageWithMetadata}
-				isi.Namespace = kapi.NamespaceValue(ctx)
-				isi.Name = id
-				return &isi, nil
+			// TODO: trim based on a better algorithm
+			base := strings.TrimPrefix(tagging.Image, "sha256:")
```
Comment (ncdc) [inline pkg/image/registry/imagestreamimage/rest.go]: You can try to use `digest.ParseDigest`. If it succeeds, there are methods to get just the algorithm and just the hex parts separately. If it fails, it's not a v2 digest.

**Comment by ncdc** [comment]
This changes ISI.Name so instead of being

    origin-ruby-sample@sha256:a2f15cc10423c165ca221f4a7beb1f2949fb0f5acbbc8e3a0250eb7d5593ae64

it's instead

    origin-ruby-sample@a2f15cc

correct?

**Comment by ncdc** [comment]
I'd make the digest.ParseDigest change I mentioned above in the 2 places where you trim off the sha256: prefix. Otherwise LGTM

**Comment by smarterclayton** [comment]
Yes.  Right now I'm neglecting collisions, but we can easily determine the collision minimum length at the point in time we return the name by traversing the tag structure, so it's easy to fix later.  Also, we don't have a million tags so the risk of collisions is low.

----- Original Message -----

**Comment by smarterclayton** [comment]
Made those changes

----- Original Message -----

**Comment by ncdc** [comment]
LGTM [merge]

**Comment by smarterclayton** [comment]
[test]

**Pull request merged**